### PR TITLE
Aquaria: Logic bug fixes

### PR DESCRIPTION
**Pull request by tioui**
## What is this fixing or adding?

In the last async, some logic bugs has been identified. So I fixed those bugs so that the logic does not failed.

There is also two simple location renaming because they did not make any sens.

## How was this tested?

The logic has been tests with 3 steps.
 - First, the unit tests has been used to validate some progressive items requirements.
 - Then, I tried to generate a lots of game with a lots of different options.
 - Finally, I run 2 games, one multiworld (with 2 Aquaria worlds) and the other single world, and using the Universal tracker to validate the accessible locations with differents progression items obtained.

### CODE REVIEW

File: worlds/aquaria/Regions.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 def _has_hot_soup(state:CollectionState, player: int) -> bool:
     """`player` in `state` has the hotsoup item"""
-    return state.has("Hot soup", player)
+    return state.has_any ({"Hot soup", "Hot soup x 2"}, player)
```
Comment (Exempt-Medic) [inline worlds/aquaria/Regions.py:19]: Does this really work with the space?
```suggestion
    return state.has_any({"Hot soup", "Hot soup x 2"}, player)
```

**Review by Exempt-Medic — APPROVED** [review]
The changes all LGTM. I won't claim to know _what_ the `Regions.py` changes are, but the location names all seem to have been changed everywhere they occur and the new tests make the changes to the old test make sense. A lot of the `Regions.py` changes are pretty straightforward and I'll just trust the world dev that the others makes sense for this game.  All the items that were added into checks seem to be progression to me, so that's good. Just one small comment on a space.

File: worlds/aquaria/Regions.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 def _has_hot_soup(state:CollectionState, player: int) -> bool:
```
Comment (agilbert1412) [inline worlds/aquaria/Regions.py:17]: There's a bunch of missing spaces before the word "CollectionState" in the methods of this file.
You could run a reformat file and get it all cleaned up at once

**Review by agilbert1412 — APPROVED** [review]
PR appears to do what it claims, and I did not spot any errors.
Same as Medic, I don't know the game so I cannot assess if all the decisions are **correct**, just that they appear to do what they intend.

I have one general suggestion though, and it would be to place all these region, location and item names into string constants. That would allow a future renaming of one of them to only be done in one place, and be confident that no other usages were missed.

Not for this PR, as that would be out of scope, but something to think about for the long term :)

File: worlds/aquaria/Regions.py (hunk 2 above)
Comment (tioui) [reply worlds/aquaria/Regions.py:17]: I will look at that. Thank you.

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**